geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerFeatureSource.scala

```diff
@@ -16,124 +16,65 @@
-import org.geotools.filter.identity.FeatureIdImpl
 import org.geotools.geometry.jts.ReferencedEnvelope
 import org.geotools.referencing.crs.DefaultGeographicCRS
-import org.locationtech.geomesa.features.SerializationOption.SerializationOptions
-import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer
+import org.locationtech.geomesa.kafka.KafkaDataStore.FeatureSourceFactory
+import org.locationtech.geomesa.kafka.consumer.KafkaConsumerFactory
 import org.locationtech.geomesa.security.ContentFeatureSourceSecuritySupport
 import org.locationtech.geomesa.utils.geotools.ContentFeatureSourceReTypingSupport
 import org.locationtech.geomesa.utils.geotools.Conversions._
-import org.locationtech.geomesa.utils.index.{QuadTreeFeatureStore, SynchronizedQuadtree}
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes.{DFI, DFR, FR}
+import org.locationtech.geomesa.utils.index.QuadTreeFeatureStore
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
-import org.opengis.filter.identity.FeatureId
 import org.opengis.filter.spatial.{BBOX, BinarySpatialOperator, Within}
 import org.opengis.filter.{And, Filter, IncludeFilter, Or}
 
 import scala.collection.JavaConversions._
+import scala.collection.mutable
 
-class KafkaConsumerFeatureSource(entry: ContentEntry,
-                                 val sft: SimpleFeatureType,
-                                 eb: EventBus,
-                                 query: Query,
-                                 topic: String,
-                                 zookeepers: String,
-                                 expiry: Boolean,
-                                 expirationPeriod: Long)
-  extends ContentFeatureStore(entry, query)
+abstract class KafkaConsumerFeatureSource(entry: ContentEntry,
+                                          schema: SimpleFeatureType,
+                                          query: Query)
+  extends ContentFeatureSource(entry, query)
   with ContentFeatureSourceSecuritySupport
-  with ContentFeatureSourceReTypingSupport
-  with QuadTreeFeatureStore {
+  with ContentFeatureSourceReTypingSupport {
 
-  override type FR = QuadTreeFeatureStore#FR
-
-  var qt = new SynchronizedQuadtree
-
-  val groupId = RandomStringUtils.randomAlphanumeric(5)
-  val decoder = new KryoFeatureSerializer(getSchema, SerializationOptions.withUserData)
-  // create a producer that reads from kafka and sends to the event bus
-  val consumer = new KafkaFeatureConsumer(topic, zookeepers, groupId, decoder, eb)
-
-  case class FeatureHolder(sf: SimpleFeature, env: Envelope) {
-    override def hashCode(): Int = sf.hashCode()
-
-    override def equals(obj: scala.Any): Boolean = obj match {
-      case other: FeatureHolder => sf.equals(other.sf)
-      case _ => false
-    }
-  }
-
-  val cb = CacheBuilder.newBuilder()
+  override def getBoundsInternal(query: Query) =
+    ReferencedEnvelope.create(new Envelope(-180, 180, -90, 90), DefaultGeographicCRS.WGS84)
```

Comment:
Should use the shared constant for 4326 here - i think it's in simple feature types. It will use the full crs if available.